schemas/v3.0/schema.yaml

```diff
@@ -0,0 +1,1500 @@
+      parameters:
+        type: object
+        patternProperties:
+          '^[a-zA-Z0-9\.\-_]+$':
+            oneOf:
+              - $ref: '#/definitions/Reference'
+              - $ref: '#/definitions/Parameter'
+      examples:
+        type: object
+        patternProperties:
+          '^[a-zA-Z0-9\.\-_]+$':
+            oneOf:
+              - $ref: '#/definitions/Reference'
+              - $ref: '#/definitions/Example'
+      requestBodies:
+        type: object
+        patternProperties:
+          '^[a-zA-Z0-9\.\-_]+$':
+            oneOf:
+              - $ref: '#/definitions/Reference'
+              - $ref: '#/definitions/RequestBody'
+      headers:
+        type: object
+        patternProperties:
+          '^[a-zA-Z0-9\.\-_]+$':
+            oneOf:
+              - $ref: '#/definitions/Reference'
+              - $ref: '#/definitions/Header'
+      securitySchemes:
+        type: object
+        patternProperties:
+          '^[a-zA-Z0-9\.\-_]+$':
+            oneOf:
+              - $ref: '#/definitions/Reference'
+              - $ref: '#/definitions/SecurityScheme'
+      links:
+        type: object
+        patternProperties:
+          '^[a-zA-Z0-9\.\-_]+$':
+            oneOf:
+              - $ref: '#/definitions/Reference'
+              - $ref: '#/definitions/Link'
+      callbacks:
+        type: object
+        patternProperties:
+          '^[a-zA-Z0-9\.\-_]+$':
+            oneOf:
+              - $ref: '#/definitions/Reference'
+              - $ref: '#/definitions/Callback'
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false
+
+  Schema:
+    type: object
+    properties:
+      title:
+        type: string
+      multipleOf:
+        type: number
+        minimum: 0
+        exclusiveMinimum: true
+      maximum:
+        type: number
+      exclusiveMaximum:
+        type: boolean
+        default: false
+      minimum:
+        type: number
+      exclusiveMinimum:
+        type: boolean
+        default: false
+      maxLength:
+        type: integer
+        minimum: 0
+      minLength:
+        type: integer
+        minimum: 0
+        default: 0
+      pattern:
+        type: string
+        format: regex
+      maxItems:
+        type: integer
+        minimum: 0
+      minItems:
+        type: integer
+        minimum: 0
+        default: 0
+      uniqueItems:
+        type: boolean
+        default: false
+      maxProperties:
+        type: integer
+        minimum: 0
+      minProperties:
+        type: integer
+        minimum: 0
+        default: 0
+      required:
+        type: array
+        items:
+          type: string
+        minItems: 1
+        uniqueItems: true
+      enum:
+        type: array
+        items: {}
+        minItems: 1
+        uniqueItems: true
+      type:
+        type: string
+        enum:
+          - array
+          - boolean
+          - integer
+          - number
+          - object
+          - string
```

Comment:
Would it be possible to get `"null"` added to this list? We return some properties that are present in a response object, but contain a `null` value, and we need to have the type `null` to accurate describe them.

Reply:
Ah, I see now that 3.0.0 indicates that it doesn't support the `null` type.

Do you have any feeling as to whether this is set in stone yet? Unless I'm really misunderstanding something, the trouble with `nullable` is that it's not in the JSON schema core, and means that we can no longer use off-the-shelf JSON schema validators to validate the schemas that are generated for OpenAPI 3.0.

Reply:
And alright, I just read through https://github.com/OAI/OpenAPI-Specification/issues/229.

It doesn't look like the core team can be persuaded on this point, but oof, one of my perceived advantages of the 3.0 upgrade was not having to deal with the arbitrary restrictions that 2.0 imposed on JSON schemas. It seems that there are now fewer of them for 3.0, but they're still there, and JSON schema generated to be OpenAPI-compatible still cannot be used to accurately describe all types of API resources without additional meta-annotations (which are obviously not portable).

I'm not sure where that leaves us.

Reply:
V3 leaves us in the position where the Schema object can processed with JSON schema tooling without generating errors (due mainly to type: file).  It is true that not everything in JSON schema is supported.  Hopefully we can get there in the future.

Reply:
@brandur see also #1389